Fix issue with Throttle and Trace adaptors losing Requester trait bounds

pfa230 opened the pull request:
The issue is rather annoying and seems like it reoccurs from time to time. I added a (rather clumsy) test to prevent it from getting back.

WaffleLapkin commented:
@pfa230 do you think a test here helps? seems like we are just as likely to forget to add a new adaptor to the test.

pfa230 commented:
> @pfa230 do you think a test here helps? seems like we are just as likely to forget to add a new adaptor to the test.

The test is meant to fail compilation whenever an issue like this is introduced. I added it in a separate commit, prior to fixing the issue itself, specifically to allow replicating this:
- [Add test for adaptors losing Requester trait bounds issue](https://github.com/teloxide/teloxide/pull/1105/commits/93782fba10bdce0a911d3a2101c07e8360dc6dc6)
- Run ```$ cargo test --all-features -p teloxide-core --lib```

The expected result is this:
```
error[E0599]: the method `erase` exists for struct `DefaultParseMode<Trace<CacheMe<Throttle<Bot>>>>`, but its trait bounds were not satisfied
   --> crates/teloxide-core/src/adaptors.rs:79:14
    |
74  |           let _ = bot
    |  _________________-
75  | |             .throttle(Limits::default())
76  | |             .cache_me()
77  | |             .trace(Settings::empty())
78  | |             .parse_mode(ParseMode::MarkdownV2)
79  | |             .erase();
    | |             -^^^^^ method cannot be called on `DefaultParseMode<Trace<CacheMe<Throttle<Bot>>>>` due to unsatisfied trait bounds
    | |_____________|
```

I'm not sure how CI is setup, and if it doesn't run tests with ```--all-features``` - the test is likely won't trigger. But in this case I would argue it would make sense to add ```--all-features``` to CI.

WaffleLapkin commented:
> The test is meant to fail compilation whenever an issue like this is introduced. 

@pfa230 my question is: how would we introduce such an issue with an existing adaptor? I don't really see anyone removing `#[derive(Clone)]` or w/e.

I would expect such issues if we added a new adaptor, but then we are just as likely to not add it to this test I think...

pfa230 commented:
> @pfa230 my question is: how would we introduce such an issue with an existing adaptor? I don't really see anyone removing `#[derive(Clone)]` or w/e.
> 
> I would expect such issues if we added a new adaptor, but then we are just as likely to not add it to this test I think...

Good point, removed the test.
